packages/lit-dev-tools-esm/src/build-unversioned-docs.ts

```diff
@@ -0,0 +1,173 @@
+  const relativeChildPathWithoutExt = relativeChildPath.slice(0, -ext.length);
+  const fileBasename = pathlib.basename(relativeChildPath, ext);
+
+  let fileContents = fs.readFileSync(path, {encoding: 'utf8'});
+  let unversionedLocation = pathlib.join(
+    UNVERSIONED_VERSION_LOCATION,
+    relativeChildPath
+  );
+  switch (ext) {
+    case '.md':
+      const [frontMatterData, restOfFile] = getFrontMatterData(fileContents);
+      const hasExistingPermalink = frontMatterData.find((val) =>
+        val.includes('permalink:')
+      );
+      if (hasExistingPermalink) {
+        throw new Error(
+          `Unhandled case: Handle existing permalink in '${path}'`
+        );
+      }
+      if (fileBasename === 'index') {
+        frontMatterData.push(
+          `permalink: docs/${relativeChildPathWithoutExt}.html`
+        );
+      } else {
+        frontMatterData.push(
+          `permalink: docs/${relativeChildPathWithoutExt}/index.html`
+        );
+      }
+
+      fileContents = writeFrontMatter(frontMatterData) + restOfFile;
+      break;
+    case '.json':
+      // Only handle the latest version json file. E.g., `v[N].json` which needs
+      // to match its directory name. Therefore the latest json file will need
+      // to be renamed to `unversioned.json` as it is being moved to the
+      // `unversioned` subdirectory. All other json files will continue to match
+      // their directory name.
+      if (fileBasename === SITE_LATEST_VERSION) {
+        unversionedLocation = pathlib.join(
+          pathlib.dirname(unversionedLocation),
+          'unversioned.json'
+        );
+        fileContents = JSON.stringify({
+          collection: 'docs-unversioned',
+          latestVersion: SITE_LATEST_VERSION,
+        });
+      }
+      break;
+    case '.html':
+      // api.html is an html file containing only front matter data used to
+      // generate the API documentation. It contains the versioned api base
+      // paths. This transform makes the api base path unversioned.
+      if (fileBasename === 'api') {
+        const [frontMatterData] = getFrontMatterData(fileContents);
+        const transformedFrontMatter = frontMatterData.map((line) => {
+          return line.replace(`/${SITE_LATEST_VERSION}/`, '/');
+        });
+        fileContents = writeFrontMatter(transformedFrontMatter);
+      } else {
+        throw new Error(`Unhandled html document: '${path}'`);
+      }
+      break;
+    default:
+      throw new Error(`Unhandled extension '${ext}' for '${path}'`);
+  }
+
+  console.log(
+    `Writing: ${pathlib.relative(
+      CONTENT_PKG,
+      unversionedLocation
+    )} from ${pathlib.relative(CONTENT_PKG, path)}`
+  );
+  fs.mkdirSync(pathlib.dirname(unversionedLocation), {recursive: true});
+  fs.writeFileSync(unversionedLocation, fileContents);
+}
+
+/**
+ * Retrieve the 11ty frontmatter from our docs. This is not a full fledged YAML
+ * parser.
+ */
+function getFrontMatterData(
```

Comment:
Why not use a frontmatter parser libraray?

Reply:
I did not want to add a dependency on a YAML parser for such scoped and light work.